### PR TITLE
Fix trainer seq2seq qa.py evaluate log and ft script

### PR DESCRIPTION
**Pull request by iamtatsuki05**
# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sgugger  @patil-suraj

### CODE REVIEW

File: examples/pytorch/question-answering/run_seq2seq_qa.py [hunk 1]
```diff
@@ -476,15 +483,18 @@ def preprocess_validation_function(examples):
...
-            return_overflowing_tokens=True,
             return_offsets_mapping=True,
+            # return_overflowing_tokens=True,
         )
-        # Tokenize targets with the `text_target` keyword argument
-        labels = tokenizer(text_target=targets, max_length=max_answer_length, padding=padding, truncation=True)
+
+        # Setup the tokenizer for targets
+        with tokenizer.as_target_tokenizer():
+            labels = tokenizer(targets, max_length=max_answer_length, padding=padding, truncation=True)
```
Comment (sgugger) [inline examples/pytorch/question-answering/run_seq2seq_qa.py:492]: No, this API is deprecated. Please revert this change.

File: examples/pytorch/question-answering/run_seq2seq_qa.py [hunk 2]
```diff
@@ -476,15 +483,18 @@ def preprocess_validation_function(examples):
             max_length=max_seq_length,
             padding=padding,
             truncation=True,
-            return_overflowing_tokens=True,
             return_offsets_mapping=True,
+            # return_overflowing_tokens=True,
```
Comment (sgugger) [inline examples/pytorch/question-answering/run_seq2seq_qa.py:487]: This shouldn't be commented out.

File: examples/pytorch/question-answering/run_seq2seq_qa.py [hunk 3]
```diff
@@ -476,15 +483,18 @@ def preprocess_validation_function(examples):
...
-        labels = tokenizer(text_target=targets, max_length=max_answer_length, padding=padding, truncation=True)
+
+        # Setup the tokenizer for targets
+        with tokenizer.as_target_tokenizer():
+            labels = tokenizer(targets, max_length=max_answer_length, padding=padding, truncation=True)
 
         # Since one example might give us several features if it has a long context, we need a map from a feature to
         # its corresponding example. This key gives us just that.
-        sample_mapping = model_inputs.pop("overflow_to_sample_mapping")
+        # sample_mapping = model_inputs.pop("overflow_to_sample_mapping")
```
Comment (sgugger) [inline examples/pytorch/question-answering/run_seq2seq_qa.py:496]: Same here.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! This removes a lot of functionality from the example and reverts some unwanted changes however, so this should be fixed before we can merge.

File: examples/pytorch/question-answering/run_seq2seq_qa.py (hunk 1 above)
Comment (iamtatsuki05) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:492]: I see

File: examples/pytorch/question-answering/run_seq2seq_qa.py (hunk 2 above)
Comment (iamtatsuki05) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:487]: I think I'm going to keep it because this problem https://github.com/huggingface/transformers/issues/15398

File: examples/pytorch/question-answering/run_seq2seq_qa.py (hunk 3 above)
Comment (iamtatsuki05) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:496]: The same problem as above occurs.

File: examples/pytorch/question-answering/run_seq2seq_qa.py (hunk 2 above)
Comment (sgugger) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:487]: This is not a fix of the issue, just a downgrade of the example.

Comment (iamtatsuki05) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:487]: This is an inherent problem, so I will take it back.

File: examples/pytorch/question-answering/run_seq2seq_qa.py [hunk 4]
```diff
@@ -476,10 +483,11 @@ def preprocess_validation_function(examples):
             max_length=max_seq_length,
             padding=padding,
             truncation=True,
-            return_overflowing_tokens=True,
             return_offsets_mapping=True,
+            return_overflowing_tokens=True,
         )
-        # Tokenize targets with the `text_target` keyword argument
+
+        # Setup the tokenizer for targets
```
Comment (sgugger) [inline examples/pytorch/question-answering/run_seq2seq_qa.py:490]: The comment should be the one deleted in the diff and not the one added :-)

**Review by sgugger — APPROVED** [review]
Thanks! Just one last nit and we should be good to merge!

File: examples/pytorch/question-answering/run_seq2seq_qa.py (hunk 4 above)
Comment (iamtatsuki05) [reply examples/pytorch/question-answering/run_seq2seq_qa.py:490]: sorry, revert to original version

**Comment by iamtatsuki05** [comment]
Synchronized data from the main branch.
